### ISSUE TITLE
Page 'onunload' and 'onbeforeunload' - Prevents Block

### ISSUE THREAD

**Issue by HeyVern** [proginosko/LeechBlockNG#386]
Certain webpages use 'onunload' and 'onbeforeunload' to present such messages as "are you sure you want to leave this page"...etc.  The issue is, if this is a page you are trying to block, the user can simple press 'cancel' and the page navigation to the block page is canceled effectively allowing the user to bypass the block until the page is manually refreshed.  Is there any way to bypass such prompts and force the page to refresh anyway, or simply block the script execution by filtering it out?

Example page: taming.io

Thanks!

**Comment by smayer97** [comment]
sounds like this is the same or similar problem as presented for the Chrome version here: 
https://github.com/proginosko/LeechBlockNG-chrome/issues/61

**Comment by proginosko** [comment]
Do you have the option "Immediately block pages..." selected?

**Comment by HeyVern** [comment]
Yes.  Works on every page except those that use onunload or onbeforeunload to present a dialog box.

**Comment by proginosko** [comment]
This is an interesting problem to solve, at least on Firefox. Every time LB tries to block a page with an _onbeforeunload_ listener, the "Leave page"/"Stay on page" dialog box pops up, but LB keeps trying to block the page, resulting in an ever-growing stack of dialog boxes (one for each attempt to block!). You can keep clicking "Stay on page" but LB will keep trying to block it, until you eventually click "Leave page". I'm not sure it's possible for an extension to bypass or disable the _onbeforeunload_ listener, but I will keep investigating.

In the meantime, one workaround for Firefox is to completely disable the _beforeunload_ event:

https://support.mozilla.org/en-US/questions/1338923

**Comment by HeyVern** [comment]
[This site](https://superuser.com/questions/705307/how-can-i-disable-are-you-sure-you-want-to-leave-this-page-popups-in-chrome) discusses the topic and suggests inserting _`$(window).off('beforeunload.windowReload');`_ into the page.  It should disable the popups.

**Comment by proginosko** [comment]
Thanks for doing some research on this. The downside of this solution is that it requires the jQuery library, and while LB uses that for its GUI, it wouldn't really be good practice to inject that entire library into a web page just to remove one event listener! I will continue to investigate alternative solutions.

**Comment by HeyVern** [comment]
Any news on this particular issue?  Thanks!
